CenterForOpenScience/osf.io
# [Feature] Fix prereg checkouts [OSF-8273]

## Purpose
Deleting a prereg draft that a prereg admin viewed leaves files checked out to admins.


## Changes
Check in pre-reg related files when draft is deleted.
BEFORE:
![screen shot 2017-08-01 at 4 08 43 pm](https://user-images.githubusercontent.com/1322421/28844759-bbb91c68-76d3-11e7-8d70-ff504765b1e3.png)
AFTER:
<img width="1039" alt="screen shot 2017-08-01 at 4 04 29 pm" src="https://user-images.githubusercontent.com/1322421/28844766-c4916f02-76d3-11e7-827e-7520c4eb9e83.png">



## Side effects
Disable the ability for users to force check in files checked out to prereg admins (Per discussions with BrianG, David Mellor, and Alex DeHaven)



## Ticket

https://openscience.atlassian.net/browse/OSF-8273

## QA Notes
At a minimum:
1. Start a draft registration and select  a file. Do not finish. Save draft. Check out file. Delete draft. File should still be checked out to you
2. Create a prereg (with files) and submit it. As prereg admin, view the draft. Confirm files are checked out and there is no force check in button (project admin). Delete draft as project admin. Files should no longer be checked out.
3. Confirm that a prereg can be approved and files still get checked back in
